File: nodejs/src/contacts_example.js
const MailproClient = require('./mailproClient');

(async () => {
  try {
    const client = new MailproClient('YOUR_API_KEY');

    // Create a new contact
    const newContact = {
      email: 'newcontact@example.com',
      firstName: 'John',
      lastName: 'Doe'
    };
    const createRes = await client.createContact(newContact);
    console.log('Contact created:', createRes);

    // Fetch all contacts
    const contacts = await client.getContacts();
    console.log('All contacts:', contacts);
  } catch (err) {
    console.error('Error managing contacts:', err.message);
  }
})();

File: python/contacts_example.py
const MailproClient = require('./mailproClient');

(async () => {
  const token = process.env.MAILPRO_TOKEN;
  const client = new MailproClient({ token });
  try {
    // Create
    const newContact = { email: 'new@example.com', firstName: 'New', lastName: 'Contact' };
    const created = await client.createContact(newContact);
    console.log('Created contact:', created);

    // Update
    const contactId = created && created.id ? created.id : null;
    if (contactId) {
      const updated = await client.updateContact(contactId, { lastName: 'Updated' });
      console.log('Updated contact:', updated);

      // Delete
      // const del = await client.deleteContact(contactId);
      // console.log('Deleted contact:', del);
    }
  } catch (err) {
    console.error('Contacts error:', err.message);
  }
})();
